### PR TITLE
samples / fixing authorize dataset logic by updating source dataset acl 

### PR DESCRIPTION
**Pull request by nikunjbhartia**
Fixes#2188

Fixed the logic for authorizing a bigquery user dataset but updating the source dataset acl with that of the user dataset information.

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/bigquery/AuthorizeDataset.java [hunk 1]
```diff
@@ -58,12 +58,12 @@ public static void authorizeDataset(DatasetId sourceDatasetId, DatasetId userDat
           new Acl.DatasetAclEntity(userDatasetId, targetTypes);
       sourceDatasetAcl.add(Acl.of(userDatasetAclEntity));
 
-      // update the user dataset with source dataset's ACL
-      Dataset updatedUserDataset =
-          userDataset.toBuilder().setAcl(sourceDatasetAcl).build().update();
+      // update the source dataset with user dataset's ACL
```
Comment (averikitsch) [inline samples/snippets/src/main/java/com/example/bigquery/AuthorizeDataset.java:61]: The method comment says: "This method will update userDataset's ACL with sourceDataset's ACL". Does this need to be updated as well or is this the right change?

Comment (nikunjbhartia) [reply samples/snippets/src/main/java/com/example/bigquery/AuthorizeDataset.java:61]: Hi @averikitsch Thanks for pointing out ! 
I have updated the method. 
And yes, in this case, source dataset ACL should be updated (a similar case for authorizing a view can be found here https://github.com/googleapis/java-bigquery/blob/HEAD/samples/snippets/src/main/java/com/example/bigquery/AuthorizedViewTutorial.java#L99 )

**Review by averikitsch — APPROVED** [review]

**Comment by nikunjbhartia** [comment]
Updated the failing test case for authorized Dataset.

**Comment by nikunjbhartia** [comment]
Hi, seems this is pending review from one of the owners. Is there something I am missing ?

**Comment by averikitsch** [comment]
@googleapis/api-bigquery and @googleapis/yoshi-java PTAL

**Pull request merged**